src/Algebra/Properties/CommutativeSemigroup.agda

```diff
@@ -0,0 +1,121 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some theory for commutative semigroup
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra using (Semigroup)
+open import Algebra.FunctionProperties using (Commutative)
+
+module Algebra.Properties.CommutativeSemigroup {ℓ ℓ=} (H : Semigroup ℓ ℓ=)
+                                     (open Semigroup H using (_≈_; _∙_))
+                                     (comm : Commutative _≈_ _∙_)
+                                     where
+
+import      Relation.Binary.Properties.Setoid as OfSetoid
+open import Function using (_$_)
+import      Relation.Binary.EqReasoning as EqR
+
+open Semigroup H using (refl; sym; trans; assoc; ∙-congˡ; ∙-congʳ; setoid)
+open OfSetoid setoid public
+open EqR setoid
```

Comment:
The previous two `open` statements can be combined on one line with the `import` statements.

Reply:
Done.

Reply:
Sorry, I have missed something. Now I change this to
```
open Semigroup H public hiding (_≈_; _∙_)
open EqR setoid
```
``hiding`` set in order to avoid clash with the import in the module decl.